src/Core/Application/Catalog/Services/BrandService.cs

```diff
@@ -29,7 +29,7 @@ public async Task<Result<Guid>> CreateBrandAsync(CreateBrandRequest request)
         if (brandExists) throw new EntityAlreadyExistsException(string.Format(_localizer["brand.alreadyexists"], request.Name));
         var brand = new Brand(request.Name, request.Description);
         brand.DomainEvents.Add(new StatsChangedEvent());
-        var brandId = await _repository.CreateAsync(brand);
+        var brandId = await _repository.CreateAsync<Brand, Guid>(brand);
```

Comment:
so, this means that I would have to know what kind of primary key Brand has while developing? What's expected to happen if I pass int instead of Guid?

Reply:
yeah, developer must know which type must be used.
and it seems that [C# Method type inference ignores generic constraints on the method type parameters](https://stackoverflow.com/a/6630769/908464) so language level has limitations.
But the good news is compiler will mark code as error e.q.
`There is no implicit reference conversion from 'DN.WebApi.Domain.Catalog.Product' to 'DN.WebApi.Domain.Common.Contracts.BaseEntityWith<int>'`

Reply:
Also, we can provide a pre-configured with Guid or int as `TKey`
for example:
```c#
Task UpdateAsync<T, TKey>(T entity, CancellationToken cancellationToken = default)
where T : BaseEntityWith<TKey>;

Task UpdateAsync<T>(T entity, CancellationToken cancellationToken = default)
where T : BaseEntityWith<Guid>;
```
although I'm in favor of `int` being default key type

Reply:
@fretje yeah I don't like `BaseEntityWith` either. But `BaseEntity : BaseEntity<Guid>` is an entirely new class and when we inherited `AuditableEntity` from `BaseEntity<Guid>` there is no direct relation between `AuditableEntity` and `BaseEntity` and we got compiler error. Hence we need a common base class.

Reply:
I don't see the problem. AuditableEntity needs to be changed to AuditableEntity<T> which derives from BaseEntity<T>, then a new non generic AuditableEntity class can be created which is derived from AuditableEntity<defaultIdType>, and again backwards compatible...